base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -753,6 +769,95 @@ where
+            .select_utxos(
+                total_split_amount,
+                fee_per_gram,
+                output_count,
+                UTXOSelectionStrategy::MaturityThenSmallest,
+            )
+            .await?;
+        let utxo_total = inputs.iter().fold(MicroTari::from(0), |acc, x| acc + x.value);
+        let input_count = inputs.len();
+        if require_change_output {
+            output_count = split_count + 1
+        };
+        let fee = Fee::calculate(fee_per_gram, 1, input_count, output_count);
+
+        trace!(target: LOG_TARGET, "Construct coin split transaction.");
+        let offset = PrivateKey::random(&mut OsRng);
+        let nonce = PrivateKey::random(&mut OsRng);
+        let mut builder = SenderTransactionProtocol::builder(0);
+        builder
+            .with_lock_height(lock_height.unwrap_or(0))
+            .with_fee_per_gram(fee_per_gram)
+            .with_offset(offset.clone())
+            .with_private_nonce(nonce.clone());
+        trace!(target: LOG_TARGET, "Add inputs to coin split transaction.");
+        for uo in inputs.iter() {
+            builder.with_input(
+                uo.as_transaction_input(&self.factories.commitment, uo.clone().features),
+                uo.clone(),
+            );
+        }
+        trace!(target: LOG_TARGET, "Add outputs to coin split transaction.");
+        let mut outputs = Vec::with_capacity(output_count);
+        let change_output = utxo_total
+            .checked_sub(fee)
+            .ok_or(OutputManagerError::NotEnoughFunds)?
+            .checked_sub(total_split_amount)
+            .ok_or(OutputManagerError::NotEnoughFunds)?;
+        for i in 0..output_count {
+            let output_amount = if i < split_count {
+                amount_per_split
+            } else {
+                change_output
+            };
+
+            let mut spend_key = PrivateKey::default();
+            {
+                let mut km = acquire_lock!(self.key_manager);
+                spend_key = km.next_key()?.k;
+            }
+            self.db.increment_key_index().await?;
+            let utxo = UnblindedOutput::new(output_amount, spend_key, None);
+            outputs.push(utxo.clone());
+            builder.with_output(utxo);
+        }
+        trace!(target: LOG_TARGET, "Build coin split transaction.");
+        let factories = CryptoFactories::default();
+        let mut stp = builder
+            .build::<HashDigest>(&self.factories)
+            .map_err(|e| OutputManagerError::BuildError(e.message))?;
+        // The Transaction Protocol built successfully so we will pull the unspent outputs out of the unspent list and
+        // store them until the transaction times out OR is confirmed
+        let tx_id = stp.get_tx_id()?;
+        trace!(
+            target: LOG_TARGET,
+            "Encumber coin split transaction ({}) outputs.",
+            tx_id
+        );
+        self.db.encumber_outputs(tx_id, inputs, outputs).await?;
```

Comment:
Need to add the following just after this:
`self.confirm_encumberance(tx_id).await?;`

Reply:
Perfect, change applied.